src/ngx_http_lua_directive.c

```diff
@@ -1602,7 +1603,12 @@ ngx_http_lua_conf_read_lua_token(ngx_conf_t *cf,
             }
         }
 
-        b->pos += ovec[1];
+        if (ctx->expect_right_lbracket && rc == FOUND_COMMENT_LINE) {
+            b->pos += (ovec[0] + 2);
```

Comment:
Hmm, I think it's better to avoid returning `FOUND_COMMENT_LINE` in this case since we are not really seeing a comment line. Maybe directly skip the current token and try reading the next one? If that's too complicated, maybe we can just introduce a `FOUND_DONTCARE` psudo token and return that instead? Thoughts?